electrum/gui/qt/terms_and_conditions_mixin.py

```diff
@@ -0,0 +1,130 @@
+        warning_box.addWidget(icon_label)
+        warning_box.addWidget(warning_label)
+        warning_box.addStretch(1)
+        vbox.addLayout(warning_box)
+        hbox = QHBoxLayout()
+        back_button = QPushButton(_('No, I changed my mind'))
+        back_button.clicked.connect(lambda: self.close())
+        cancel_button = QPushButton(_('Yes, I disagree'))
+        cancel_button.clicked.connect(self.cancel)
+        cancel_button.setDefault(True)
+        hbox.addStretch(1)
+        hbox.addWidget(back_button)
+        hbox.addWidget(cancel_button)
+        vbox.addStretch(1)
+        vbox.addLayout(hbox)
+        self.exec_()
+
+    def cancel(self):
+        self.second_chance_for_reading = False
+        self.close()
+
+
+class TermsAndConditionsTextEdit(QTextEdit):
+    def __init__(self, next_button, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.next_button = next_button
+        self.setReadOnly(True)
+        scroll_bar = self.verticalScrollBar()
+        scroll_bar.valueChanged.connect(self.scrolled)
+        self._scroll_bar_range_hit_max = False
+        scroll_bar.rangeChanged.connect(self.range_changed)
+
+    def _scroll_bar_max_position(self):
+        return self.verticalScrollBar().maximum()
+
+    def scrolled(self, scroll_bar_position:int):
+        # check if user scrolled to the bottom of the text
+        if scroll_bar_position == self._scroll_bar_max_position():
+            self.next_button.setEnabled(True)
+
+    def range_changed(self, *range):
+        # it is called when window is resized
+        if not self._scroll_bar_range_hit_max:
+            if range[1]:
+                self.next_button.setEnabled(False)
+            else:
+                self._scroll_bar_range_hit_max = True
+                self.next_button.setEnabled(True)
+
+
+class TermsAndConditionsMixin:
+    # TODO change file with lorem ipsum
+    def _read_terms_and_conditions(self) -> str:
+        dir_ = Path(os.path.abspath(os.path.dirname(__file__))).parent.parent / 'terms_and_conditions'
+        language = self.config.get('language', 'en_UK')
+        path = dir_ / f'{language}.txt'
+        if not path.exists():
+            path = path.parent / 'en_UK.txt'
+            if not path.exists():
+                raise FileNotFoundError(f'Cannot open {path}')
+        with open(path, 'r') as file:
+            return file.read()
+
+    def _render_main_dialog(self, text, run_warning=True):
+        vbox = QVBoxLayout()
+        text_edit = TermsAndConditionsTextEdit(self.next_button)
+        text_edit.setText(text)
+        vbox.addWidget(text_edit)
+        self.next_button.setText(_('I agree'))
+        self.back_button.setText(_('I disagree'))
+        try:
+            # if user agreed with terms and conditions then result is 2
+            # otherwise GoBack exception is raised
+            result = self.exec_layout(vbox, title=_('Terms & Conditions'), next_enabled=True)
+            if result == 2:
```

Comment:
Could you make it more descriptive and remove above comment?

Reply:
what about 
```python
# pushing 'I agree` button returns 2, 'I disagree` raises GoBack exception
pushed_button = self.exec_layout(vbox, title=_('Terms & Conditions'), next_enabled=True)
if pushed_button == 2:
```